src/DIRAC/Interfaces/Utilities/DCommands.py

```diff
@@ -0,0 +1,658 @@
+
+        if headers is None:
+            if len(list_) == 0:
+                return S_OK("")
+            headers = range(list_)
+
+        sortKeys = None
+        if sort is not None:
+            sortKeys = []
+            for i, s in enumerate(list_):
+                sortKeys.append((s[sort], i))
+            sortKeys.sort()
+
+        return self.fmts[self.outputFormat](list_, headers, sortKeys)
+
+    def dictFormat(self, dict_, headers=None, sort=None):
+        if headers is None:
+            headers = dict_.keys()
+        list_ = []
+        for v in dict_.values():
+            row = []
+            for h in headers:
+                row.append(v[h])
+            list_.append(row)
+
+        if sort is not None:
+            sort = headers.index(sort)
+
+        return self.listFormat(list_, headers, sort)
+
+
+# -------------------------------
+# DCommands configuration helpers
+# -------------------------------
+
+
+class DConfig:
+    def __init__(self, configDir=None, configFilename="dcommands.conf"):
+        try:
+            self.config = ConfigParser(allow_no_value=True)
+        except TypeError:
+            self.config = ConfigParser()
+
+        if not configDir:
+            var = "DCOMMANDS_CONFIG_DIR"
+            if var in os.environ:
+                configDir = os.environ[var]
+            else:
+                configDir = os.path.expanduser("~/.dirac")
+
+        self.configDir = configDir
+        self.configFilename = configFilename
+        self.configPath = os.path.join(self.configDir, self.configFilename)
+        self.bootstrapFile()
+        self.__buildSectionsAliases()
+
+    def bootstrapFile(self):
+        if not os.path.exists(self.configDir):
+            os.mkdir(self.configDir)
+            os.chmod(self.configDir, stat.S_IRWXU)
+        elif not os.path.isdir(self.configDir):
+            critical('"%s" config dir is not a directory' % self.configDir)
+        elif not os.stat(self.configDir).st_mode != stat.S_IRWXU:
+            critical('"%s" config dir doesn\'t have correct permissions' % self.configDir)
+
+        if os.path.isfile(self.configPath):
+            self.config.read(self.configPath)
+
+    def __buildSectionsAliases(self):
+        self.sectionsAliases = {}
+        for section in self.config.sections():
+            if self.config.has_option(section, "aliases"):
+                aliases = self.config.get(section, "aliases")
+                for alias in aliases.split(","):
+                    self.sectionsAliases[alias.strip()] = section
+
+        for section in self.config.sections():
+            self.sectionsAliases[section] = section
+
+    def sectionAliasName(self, alias):
+        if alias not in self.sectionsAliases:
+            return S_ERROR("DConfig section alias unknown: %s" % alias)
+
+        return S_OK(self.sectionsAliases[alias])
+
+    def write(self):
+        file_ = open(self.configPath, "w")
+        self.config.write(file_)
+        file_.close()
```

Comment:
This looks quite python 2

Reply:
It seems to pass the python3 checks, so for the time being I would prefer to leave it in.

Reply:
Nevermind Python 3, this looks quite Python 2.4 😉 (and even then it was buggy as it should have be using `try...finally`).

```suggestion
    def write(self):
        with open(self.configPath, "w") as fh:
            self.config.write(fh)
```

(The Python 3 way would be to use `pathlib`.)